salt/roster/cache.py

```diff
@@ -18,38 +60,52 @@
       - ipv4-local    # loopback addresses
 
 
-Various other keys are also supported:
+This is the default ``roster_order``.
+It prefers IPv6 over IPv4 addresses and private addresses over public ones.
+The relevant data will be fetched from the cache in-order, and the first match will fill the ``host`` key.
+
+Other address selection parameters are also possible:
+
 .. code-block:: yaml
 
-# Examples
 roster_order:
   host:
-    - id                             # use the minion id
     - global|public|private|local    # Both IPv6 and IPv4 addresses in that range
-    - grain: fqdn_ip4                # lookup this grain
-    - pillar:                        # Works with pillardata too
-        - dns:fqdn                   # ':' syntax and lists supported
-    - mine: network.ip_addrs         # Mine data lookup works the same
-    - 2000::/3                       # CIDR networks, both IPv4 and IPv6 supported
+    - 2000::/3                       # CIDR networks, both IPv4 and IPv6 are supported
 
-                                     # The lookup can be used for arbitrary other keys as well,
-  user: pillar: ssh:auth:user
-  priv:
-    - pillar: ssh:auth:private_key
 
+Using cached data
+=================
 
-.. versionadded:: Nitrogen
-    expanded functionality to include arbitrary pillar and grain keys, IPv6 addresses
-    allow roster_order per key in config
-    changed default order in line with os defaults
+Several cached libraries can be selected using the ``library: `` prefix, followed by the library key.
+This can be referenced using the same ``:`` syntax as e.g. ``pillar.get()``.
+Lists of references are also supported during the lookup.
+
+This should be especially useful for the other roster keys:
+
+.. code-block:: yaml
 
+roster_order:
+  host:
+    - grain: fqdn_ip4                # Lookup this grain
+    - mine: network.ip_addrs         # Mine data lookup works the same
+
+  user:
+    - pillar: ssh:auth:user          # Lookup this pillar key
+    - env: USER                      # Lookup this env var
```

Comment:
What good would the additional wrapper do in this scenario? What wrapper would one actually need on top of `os.environ` which is and has been stable & simple AFAICR. Not only that but `sdb://osenv/VAR` wraps around... `os.environ.get()`. I'm all for DRY but I think KISS trumps it in this case ;)

Secondly, I'm not 'using' Salt, I'm changing it; having an `sdb://` to access the env is fine from a usability point of view *for the user*. Within the salt codebase it's entirely not uncommon for `os.environ` to be directly called and it should be immediately obvious there's no way that's all going to use `sdb://` for a lot of different reasons and because `os.environ` is perfectly acceptable standard Python.